# In this notebook, we enter the raw data and save it in .csv files 

## Steam Games file

### Reading data

The `lines=True` parameter is needed because of the format of the .json file.

In [1]:
import pandas as pd
fn = 'RawData/steam_games.json.gz'
df = pd.read_json(fn, lines=True, compression='gzip')
df

publisher                                 genres  \
0                  None                                   None   
1                  None                                   None   
2                  None                                   None   
3                  None                                   None   
4                  None                                   None   
...                 ...                                    ...   
120440  Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441           Sacada              [Casual, Indie, Strategy]   
120442     Laush Studio            [Indie, Racing, Simulation]   
120443         SIXNAILS                        [Casual, Indie]   
120444             None                                   None   

                        app_name                     title  \
0                           None                      None   
1                           None                      None   
2                           None                      None   
3                           None                      None   
4                           None                      None   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                      None   

                                                      url release_date  \
0                                                    None         None   
1                                                    None         None   
2                                                    None         None   
3                                                    None         None   
4                                                    None         None   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...         None   

                                                     tags  \
0                                                    None   
1                                                    None   
2                                                    None   
3                                                    None   
4                                                    None   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url  discount_price  \
0                                                    None             NaN   
1                                                    None             NaN   
2                                                    None             NaN   
3                                                    None             NaN   
4                                                    None             NaN   
...                                                   ...             ...   
120440  http://steamcommunity.com/app/773640/reviews/?...            1.49   
120441  http://steamcommunity.com/app/733530/reviews/?...            4.24   
120442  http://steamcommunity.com/app/610660/reviews/?...            1.39   
120443  http://steamcommunity.com/app/658870/reviews/?...             NaN   
120444 

We noted that the dataframe read was partitioned in two, from index 0 to 88309 it has user items and from index 88310 to 120444 contains the data described in the [description file](https://docs.google.com/spreadsheets/d/1-t9HLzLHIGXvliq56UE_gMaWBVTPfrlTf2D9uAtLGrk/edit?usp=sharing).

In [3]:
df.loc[0:88312]

publisher                                             genres  \
0                  None                                               None   
1                  None                                               None   
2                  None                                               None   
3                  None                                               None   
4                  None                                               None   
...                 ...                                                ...   
88308              None                                               None   
88309              None                                               None   
88310         Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]   
88311  Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
88312      Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   

                      app_name                    title  \
0                         None                     None   
1                         None                     None   
2                         None                     None   
3                         None                     None   
4                         None                     None   
...                        ...                      ...   
88308                     None                     None   
88309                     None                     None   
88310      Lost Summoner Kitty      Lost Summoner Kitty   
88311                Ironbound                Ironbound   
88312  Real Pool 3D - Poolians  Real Pool 3D - Poolians   

                                                     url release_date  \
0                                                   None         None   
1                                                   None         None   
2                                                   None         None   
3                                                   None         None   
4                                                   None         None   
...                                                  ...          ...   
88308                                               None         None   
88309                                               None         None   
88310  http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
88311  http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
88312  http://store.steampowered.com/app/670290/Real_...   2017-07-24   

                                                    tags  \
0                                                   None   
1                                                   None   
2                                                   None   
3                                                   None   
4                                                   None   
...                                                  ...   
88308                                               None   
88309                                               None   
88310      [Strategy, Action, Indie, Casual, Simulation]   
88311  [Free to Play, Strategy, Indie, RPG, Card Game...   
88312  [Free to Play, Simulation, Sports, Casual, Ind...   

                                             reviews_url  discount_price  \
0                                                   None             NaN   
1                                                   None             NaN   
2                                                   None             NaN   
3                                                   None             NaN   
4                                                   None             NaN   
...                                                  ...             ...   
88308                                               None             NaN   
88309                                               None             NaN   
88310  http://steamcommunity.com/app/761140/reviews/?...            4.49   
88311  htt

### Separation of the two datasets

Detail the dataset columns

In [4]:
df.columns

Index(['publisher', 'genres', 'app_name', 'title', 'url', 'release_date',
       'tags', 'reviews_url', 'discount_price', 'specs', 'price',
       'early_access', 'id', 'metascore', 'developer', 'user_id', 'steam_id',
       'items', 'items_count'],
      dtype='object')

#### User items dataframe copy

In [5]:
df_items = df[['user_id','steam_id','items','items_count']].loc[0:88309].copy()
df_items

user_id      steam_id  \
0            76561197970982479  7.656120e+16   
1                      js41637  7.656120e+16   
2                    evcentric  7.656120e+16   
3                   Riot-Punch  7.656120e+16   
4                        doctr  7.656120e+16   
...                        ...           ...   
88305        76561198323066619  7.656120e+16   
88306        76561198326700687  7.656120e+16   
88307  XxLaughingJackClown77xX  7.656120e+16   
88308        76561198329548331  7.656120e+16   
88309        edward_tremethick  7.656120e+16   

                                                   items  items_count  
0      [{'item_id': '10', 'item_name': 'Counter-Strik...        277.0  
1      [{'item_id': '10', 'item_name': 'Counter-Strik...        888.0  
2      [{'item_id': '1200', 'item_name': 'Red Orchest...        137.0  
3      [{'item_id': '10', 'item_name': 'Counter-Strik...        328.0  
4      [{'item_id': '300', 'item_name': 'Day of Defea...        541.0  
...                                                  ...          ...  
88305  [{'item_id': '413850', 'item_name': 'CS:GO Pla...         22.0  
88306  [{'item_id': '11020', 'item_name': 'TrackMania...        177.0  
88307                                                 []          0.0  
88308  [{'item_id': '304930', 'item_name': 'Unturned'...          7.0  
88309                                                 []          0.0  

[88310 rows x 4 columns]

In [6]:
df_items.to_csv('Data/df_items.csv',index=False)

#### Steam Games dataframe copy

In [2]:
df_steam_games = df[['publisher', 'genres', 'app_name', 'title', 'url', 'release_date',
       'tags', 'reviews_url', 'discount_price', 'specs', 'price',
       'early_access', 'id', 'metascore', 'developer']].loc[88310::].copy().reset_index(drop=True)
df_steam_games

publisher                                             genres  \
0             Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]   
1      Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
2          Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
3                  彼岸领域                        [Action, Adventure, Casual]   
4                  None                                               None   
...                 ...                                                ...   
32130   Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
32131            Sacada                          [Casual, Indie, Strategy]   
32132      Laush Studio                        [Indie, Racing, Simulation]   
32133          SIXNAILS                                    [Casual, Indie]   
32134              None                                               None   

                       app_name                     title  \
0           Lost Summoner Kitty       Lost Summoner Kitty   
1                     Ironbound                 Ironbound   
2       Real Pool 3D - Poolians   Real Pool 3D - Poolians   
3                       弹炸人2222                   弹炸人2222   
4                 Log Challenge                      None   
...                         ...                       ...   
32130            Colony On Mars            Colony On Mars   
32131  LOGistICAL: South Africa  LOGistICAL: South Africa   
32132             Russian Roads             Russian Roads   
32133       EXIT 2 - Directions       EXIT 2 - Directions   
32134               Maze Run VR                      None   

                                                     url release_date  \
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
1      http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
2      http://store.steampowered.com/app/670290/Real_...   2017-07-24   
3         http://store.steampowered.com/app/767400/2222/   2017-12-07   
4      http://store.steampowered.com/app/773570/Log_C...         None   
...                                                  ...          ...   
32130  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
32131  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32134  http://store.steampowered.com/app/681550/Maze_...         None   

                                                    tags  \
0          [Strategy, Action, Indie, Casual, Simulation]   
1      [Free to Play, Strategy, Indie, RPG, Card Game...   
2      [Free to Play, Simulation, Sports, Casual, Ind...   
3                            [Action, Adventure, Casual]   
4                        [Action, Indie, Casual, Sports]   
...                                                  ...   
32130              [Strategy, Indie, Casual, Simulation]   
32131                          [Strategy, Indie, Casual]   
32132                        [Indie, Simulation, Racing]   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32134  [Early Access, Adventure, Indie, Action, Simul...   

                                             reviews_url  discount_price  \
0      http://steamcommunity.com/app/761140/reviews/?...            4.49   
1      http://steamcommunity.com/app/643980/reviews/?...             NaN   
2      http://steamcommunity.com/app/670290/reviews/?...             NaN   
3      http://steamcommunity.com/app/767400/reviews/?...            0.83   
4      http://steamcommunity.com/app/773570/reviews/?...            1.79   
...                                                  ...             ...   
32130  http://steamcommunity.com/app/773640/reviews/?...            1.49   
32131  http://steamcommunity.com/app/733530/reviews/?...            4.24   
32132  http://steamcommunity.com/app/610660/reviews/?...      

In [7]:
df_steam_games.to_csv('Data/df_steam_games.csv',index=False)

## User Reviews file

### Reading data

First, decompress the file. Note that the raw data uses single quotes (') instead of double quotes (") for the string, this is not recognized as json. So an initial transformation had to be made to open it.

In [1]:
!gzip -dk RawData/user_reviews.json.gz

In [2]:
import json
data = []
with open('RawData/user_reviews.json','r') as f:
    for l in f:
        l = l.replace("\"", "\'")
        l = l.replace("\'user_id\': \'", "\"user_id\": \"")
        l = l.replace("\', \'user_url\': \'", "\", \"user_url\": \"")
        l = l.replace("\', \'reviews\': [", "\", \"reviews\": \"[")
        l = l.replace("]}", "]\"}")
        l = l.replace("\\", "")
        
        data.append(json.loads(l))
data[0:2]

[{'user_id': '76561197970982479',
  'user_url': 'http://steamcommunity.com/profiles/76561197970982479',
  'reviews': "[{'funny': '', 'posted': 'Posted November 5, 2011.', 'last_edited': '', 'item_id': '1250', 'helpful': 'No ratings yet', 'recommend': True, 'review': 'Simple yet with great replayability. In my opinion does 'zombie' hordes and team work better than left 4 dead plus has a global leveling system. Alot of down to earth 'zombie' splattering fun for the whole family. Amazed this sort of FPS is so rare.'}, {'funny': '', 'posted': 'Posted July 15, 2011.', 'last_edited': '', 'item_id': '22200', 'helpful': 'No ratings yet', 'recommend': True, 'review': 'It's unique and worth a playthrough.'}, {'funny': '', 'posted': 'Posted April 21, 2011.', 'last_edited': '', 'item_id': '43110', 'helpful': 'No ratings yet', 'recommend': True, 'review': 'Great atmosphere. The gunplay can be a bit chunky at times but at the end of the day this game is definitely worth it and I hope they do a seque

### Convert to Dataframe and Save as .csv

In [4]:
import pandas as pd
df_reviews = pd.DataFrame(data)
df_reviews

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1                js41637               http://steamcommunity.com/id/js41637   
2              evcentric             http://steamcommunity.com/id/evcentric   
3                  doctr                 http://steamcommunity.com/id/doctr   
4              maplemage             http://steamcommunity.com/id/maplemage   
...                  ...                                                ...   
25794  76561198306599751  http://steamcommunity.com/profiles/76561198306...   
25795           Ghoustik              http://steamcommunity.com/id/Ghoustik   
25796  76561198310819422  http://steamcommunity.com/profiles/76561198310...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                                 reviews  
0      [{'funny': '', 'posted': 'Posted November 5, 2...  
1      [{'funny': '', 'posted': 'Posted June 24, 2014...  
2      [{'funny': '', 'posted': 'Posted February 3.',...  
3      [{'funny': '', 'posted': 'Posted October 14, 2...  
4      [{'funny': '3 people found this review funny',...  
...                                                  ...  
25794  [{'funny': '', 'posted': 'Posted May 31.', 'la...  
25795  [{'funny': '', 'posted': 'Posted June 17.', 'l...  
25796  [{'funny': '1 person found this review funny',...  
25797  [{'funny': '', 'posted': 'Posted July 21.', 'l...  
25798  [{'funny': '1 person found this review funny',...  

[25799 rows x 3 columns]

In [6]:
df_reviews.to_csv('Data/df_reviews.csv',index=False)

## User Items file

### Reading data

First, decompress the file. Note that the raw data uses single quotes (') instead of double quotes (") for the string, this is not recognized as json. So an initial transformation had to be made to open it.

In [13]:
!gzip -dk RawData/users_items.json.gz

In [14]:
import json
data = []
with open('RawData/users_items.json','r') as f:
    for l in f:
        l = l.replace("\"", "\'")
        l = l.replace("\'user_id\': \'", "\"user_id\": \"")
        l = l.replace("\', \'items_count\': ", "\", \"items_count\": \"")
        l = l.replace(", \'steam_id\': \'", "\", \"steam_id\": \"")
        l = l.replace("\', \'user_url\': \'", "\", \"user_url\": \"")
        l = l.replace("\', \'items\': [", "\", \"items\": \"[")
        l = l.replace("]}", "]\"}")
        l = l.replace("\\", "")
        
        data.append(json.loads(l))
data[0]

{'user_id': '76561197970982479',
 'items_count': '277',
 'steam_id': '76561197970982479',
 'user_url': 'http://steamcommunity.com/profiles/76561197970982479',
 'items': "[{'item_id': '10', 'item_name': 'Counter-Strike', 'playtime_forever': 6, 'playtime_2weeks': 0}, {'item_id': '20', 'item_name': 'Team Fortress Classic', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '30', 'item_name': 'Day of Defeat', 'playtime_forever': 7, 'playtime_2weeks': 0}, {'item_id': '40', 'item_name': 'Deathmatch Classic', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '50', 'item_name': 'Half-Life: Opposing Force', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '60', 'item_name': 'Ricochet', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '70', 'item_name': 'Half-Life', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '130', 'item_name': 'Half-Life: Blue Shift', 'playtime_forever': 0, 'playtime_2weeks': 0}, {'item_id': '300', 'item_name': 'Day of Defeat: Sou

### Convert to Dataframe and Save as .csv

In [15]:
import pandas as pd
df_user_items = pd.DataFrame(data)
df_user_items

user_id items_count           steam_id  \
0            76561197970982479         277  76561197970982479   
1                      js41637         888  76561198035864385   
2                    evcentric         137  76561198007712555   
3                   Riot-Punch         328  76561197963445855   
4                        doctr         541  76561198002099482   
...                        ...         ...                ...   
88305        76561198323066619          22  76561198323066619   
88306        76561198326700687         177  76561198326700687   
88307  XxLaughingJackClown77xX           0  76561198328759259   
88308        76561198329548331           7  76561198329548331   
88309        edward_tremethick           0  76561198331598578   

                                                user_url  \
0      http://steamcommunity.com/profiles/76561197970...   
1                   http://steamcommunity.com/id/js41637   
2                 http://steamcommunity.com/id/evcentric   
3                http://steamcommunity.com/id/Riot-Punch   
4                     http://steamcommunity.com/id/doctr   
...                                                  ...   
88305  http://steamcommunity.com/profiles/76561198323...   
88306  http://steamcommunity.com/profiles/76561198326...   
88307  http://steamcommunity.com/id/XxLaughingJackClo...   
88308  http://steamcommunity.com/profiles/76561198329...   
88309     http://steamcommunity.com/id/edward_tremethick   

                                                   items  
0      [{'item_id': '10', 'item_name': 'Counter-Strik...  
1      [{'item_id': '10', 'item_name': 'Counter-Strik...  
2      [{'item_id': '1200', 'item_name': 'Red Orchest...  
3      [{'item_id': '10', 'item_name': 'Counter-Strik...  
4      [{'item_id': '300', 'item_name': 'Day of Defea...  
...                                                  ...  
88305  [{'item_id': '413850', 'item_name': 'CS:GO Pla...  
88306  [{'item_id': '11020', 'item_name': 'TrackMania...  
88307                                                 []  
88308  [{'item_id': '304930', 'item_name': 'Unturned'...  
88309                                                 []  

[88310 rows x 5 columns]

In [16]:
df_user_items.to_csv('Data/df_user_items.csv',index=False)